src/openfunction/context.ts

```diff
@@ -196,3 +212,22 @@ export enum TraceProviderType {
    */
   OpenTelemetry = 'opentelemetry',
 }
+
+/**
+ * The specification of the component
+ * @public
+ */
+export interface ComponentSpec {
```

Comment:
[blocker] We don't need this because we only focus on `OpenFunctionContext` which is passed to the function pod from OpenFunction controller and has no relation to the state management.